src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -110,6 +125,18 @@ public void auditStarted(AuditEvent event) {
 
     @Override
     public void auditFinished(AuditEvent event) {
+        fileMessages.forEach((fileName, messages) -> {
+            writeFileOpeningTag(fileName);
```

Comment:
I don't see what the problem is without diving more into the code.
Tests can be rewritten if it is wrong.

@romani ping